### PR TITLE
Migrate `dataprotection backup-vault` to AAZ-dev-tools and add support for immutable vaults and soft delete

### PR DESCRIPTION
**Pull request by zubairabid**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az dataprotection backup-vault`

- The `dataprotection backup-vault` command was migrated to AAZ-dev-tools from autorest.
- Support for `immutability-state` flag was added.
- Support for soft delete flags was added
  - `soft-delete-state`
  - `retention-duration-in-days`/`soft-delete-retention`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

~~For new extensions:~~

~~- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).~~

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
dataprotection

**Comment by zubairabid** [comment]
Could we assign this the Jan 2023 milestone? IIUC we need to complete this PR by February 7th for that.

File: src/dataprotection/azext_dataprotection/aaz/latest/dataprotection/backup_vault/_create.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+    is_experimental=True,
+)
+class Create(AAZCommand):
+    """Create a BackupVault resource belonging to a resource group.
+
+    :example: Create BackupVault
+        az dataprotection backup-vault create --type "None" --location "WestUS" --azure-monitor-alerts-for-job-failures "Enabled" --storage-settings type="LocallyRedundant" datastore-type="VaultStore" --tags key1="val1" --resource-group "SampleResourceGroup" --vault-name "swaggerExample"
+
+    :example: Create BackupVault With MSI
+        az dataprotection backup-vault create --type "systemAssigned" --location "WestUS" --azure-monitor-alerts-for-job-failures "Enabled" --storage-settings type="LocallyRedundant" datastore-type="VaultStore" --tags key1="val1" --resource-group "SampleResourceGroup" --vault-name "swaggerExample"
```
Comment (zhoxing-ms) [inline src/dataprotection/azext_dataprotection/aaz/latest/dataprotection/backup_vault/_create.py:25]: > -  FAIL - HIGH severity: unrecognized_help_parameter_rule
    Help-Entry: `dataprotection backup-vault create` - The following parameter help names are invalid: --storage-settings

Comment (zubairabid) [reply src/dataprotection/azext_dataprotection/aaz/latest/dataprotection/backup_vault/_create.py]: The respective line has been updated, but it's still getting flagged with the same issue. As of now, the only mentions of "storage-settings" in the help text are in the `long-summary` elaborating the singular option usage:

`"Multiple actions can be specified by using more than one --storage-setting argument.\nThe --storage-settings parameter exists for backwards compatibility. The updated command is --storage-setting.\nUsage for --storage-settings: --storage-settings type=XX datastore-type=XX."`

File: src/dataprotection/azext_dataprotection/aaz/latest/dataprotection/backup_vault/_create.py [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+        tags.Element = AAZStrArg()
+
+        # define Arg Group "Properties"
+
+        _args_schema = cls._args_schema
+        _args_schema.storage_setting = AAZListArg(
+            options=["--storage-setting"],
+            singular_options=["--storage-settings"],
+            arg_group="Properties",
+            help={"short-summary": "Storage Settings. Usage: --storage-setting \"[{type:'LocallyRedundant',datastore-type:'VaultStore'}]\"", "long-summary": "Multiple actions can be specified by using more than one --storage-setting argument.\nThe --storage-settings parameter exists for backwards compatibility. The updated command is --storage-setting.\nUsage for --storage-settings: --storage-settings  type=XX datastore-type=XX."},
```
Comment (zhoxing-ms) [inline src/dataprotection/azext_dataprotection/aaz/latest/dataprotection/backup_vault/_create.py:112]: > Help-Entry: `dataprotection backup-vault create` - The following parameter help names are invalid: --storage-settings

Comment (zhoxing-ms) [reply src/dataprotection/azext_dataprotection/aaz/latest/dataprotection/backup_vault/_create.py:112]: @kairu-ms May I ask if the `singular_options` --storage-settings cannot appear in the `long-summary` now?

Comment (zubairabid) [reply src/dataprotection/azext_dataprotection/aaz/latest/dataprotection/backup_vault/_create.py]: The issue was that the prior autorest-generated help text had `--storage-settings` in its help text. Solution was to remove that.

**Review by kairu-ms — APPROVED** [review]
LGTM

**Review by evelyn-ys — CHANGES_REQUESTED** [review]
If you want to release a new version of `dataprotection` extension, please update `setup.py` and `history.rst`

**Review by evelyn-ys — APPROVED** [review]

**Comment by evelyn-ys** [comment]
@wangzelin007 We need to fix API version mismatch issue when checking min core version😂

**Comment by zubairabid** [comment]
> @wangzelin007 We need to fix API version mismatch issue when checking min core version😂

@evelyn-ys  Is there anything that I need to do to resolve this?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**